Project1/GridCell.cpp

```diff
@@ -0,0 +1,58 @@
+#include "GridCell.h"
+
+
+std::optional<CellEdge> GridCell::getEdge(GridCell* other_cell) {
+	for (CellEdge edge: _adjacent_cells) {
+		if (edge._cell == other_cell) {
+			return edge;
+		}
+	}
+	return {};
+}
+
+bool GridCell::addAdjacentCell(GridCell * new_cell)
+{
+	return addAdjacentCell(new_cell, new_cell->getTile()._terrain.getCosts());
+}
+
+/** Adds _new_cell to the adjacency vector
+	Returns true if succesful, false if _new_cell is already in the adjacency vector
+*/
+bool GridCell::addAdjacentCell(GridCell* new_cell, MobilityList<std::optional<int>> costs) {
+	if (new_cell != nullptr) {
+		if (getEdge(new_cell).has_value()) {
+			return false;
+		}			
+		_adjacent_cells.emplace_back(new_cell, costs);
+		return true;
+	}  {	return false;
+}
+}
```

Comment:
What is this spacing? Also, Idk if you have the warning on in clang-tidy, but mine says to change to this:
```
if (new_cell == nullptr || getEdge(new_cell).has_value()) {
    return false;
}
_adjacent_cells.emplace_back(new_cell, costs);
return true;
```

I would make it a void function and change it to:
```
Expects(new_cell != nullptr && !getEdge(new_cell).has_value());
_adjacent_cells.emplace_back(new_cell, costs);
```
Any other functions like that (like the one below) where the failure path is not really legitimate you should probably change to use expects as well.